### PR TITLE
Adding missing functionality of the addCardTransactionBtn

### PR DESCRIPTION
**Pull request by TatuLund**
The button that is supposed to add new invoice was non-functional, this could be how it works.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/ui-examples/30)

### CODE REVIEW

**Comment by TatuLund** [comment]
This is proposal for issues #17 and #25

File: data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java [hunk 1]
```diff
@@ -152,7 +156,14 @@ public InvoiceEditor() {
         grid.addComponentColumn(item -> createRemoveButton(grid, item))
                 .setWidth("70px").setFlexGrow(0).setTextAlign(ColumnTextAlign.CENTER);
         grid.getColumns().forEach(column -> column.setResizable(true));
-        
+
+        addCardTransactionBtn.addClickListener(event -> {
+        	ListDataProvider<Invoice> dataProvider = (ListDataProvider<Invoice>) grid.getDataProvider();
```
Comment (yuriy-fix) [inline data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java]: The logic of adding new row should be aligned with the frontend example. (Should be added at the beginning)

File: data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java [hunk 2]
```diff
@@ -174,6 +185,10 @@ public InvoiceEditor() {
         add(controlsLine, board, addsLine, grid, detailsLine);
     }
 
+    public static void scrollTo(GridPro<?> grid, int index) {
+        UI.getCurrent().getPage().executeJavaScript("$0._scrollToIndex(" + index + ")", grid.getElement());
```
Comment (yuriy-fix) [inline data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java]: We are avoiding usage of webcomponent's private API.
Moreover, it's not properly working in this case.

**Comment by yuriy-fix** [comment]
Closing in favour of https://github.com/vaadin/ui-examples/pull/42

**Pull request closed without merging**